File: 263_UglyNumber.c
//0.68%
bool isUgly(int num) {
	if(!num)
		return false;
    while(num && 0 == num % 5)
		num /= 5;
	while(num && 0 == num % 3)
		num /= 3;
	while(num && 0 == num % 2)
		num >>= 1;
	if(1 == num)
		return true;
	return false;
}

File: 263_UglyNumber.py
class Solution(object):
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """
        while num and num % 5 == 0:
            num /= 5
        while num and num % 3 == 0:
            num /= 3
        while num and num % 2 == 0:
            num /= 2
        return 1 == num

for p in 2, 3, 5:
    while num % p == 0 < num:
        num /= p
return num == 1